tests/test_api.py

```diff
@@ -245,18 +245,20 @@ def test_more_complex_deps_requires_text(self):
 
             [extra1]
             dep4
+            dep5@ git+https://example.com/python/dep.git@v1.0.0
 
             [extra2:python_version < "3"]
-            dep5
+            dep6
             """
         )
         deps = sorted(Distribution._deps_from_requires_text(requires))
         expected = [
             'dep1',
             'dep2',
-            'dep3; python_version < "3"',
-            'dep4; extra == "extra1"',
-            'dep5; (python_version < "3") and extra == "extra2"',
+            'dep3 ; python_version < "3"',
```

Comment:
Reading the spec, the name_req doesn't require the space, so I'm slightly inclined not to add it because it's allowed. I'd like to explore what it would take to add the space only where it's required (provide the minimally-satisfying syntax).

Reply:
Fixed in d8aaa72.